### PR TITLE
Helm: Add swift credentials and container name to common config

### PR DESCRIPTION
**Pull request by heytrav**
**What this PR does / why we need it**:

Add swift to common config via helm.

**Which issue(s) this PR fixes**:

No issue

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Comment by heytrav** [comment]
Note that I updated the helm/loki/CHANGELOG.md, but the version in helm/loki/Chart.yaml and helm/loki/README.md we're already incremented without a corresponding entry in the CHANGELOG.md. Please let me know if I should change my version to make everything match up.

File: production/helm/loki/CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,11 @@ Entries should be ordered as follows:
 
 Entries should include a reference to the pull request that introduced the change.
 
-[//]: # (<AUTOMATED_UPDATES_LOCATOR> : do not remove this line. This locator is used by the CI pipeline to automatically create a changelog entry for each new Loki release. Add other chart versions and respective changelog entries bellow this line.)
+[//]: # "<AUTOMATED_UPDATES_LOCATOR> : do not remove this line. This locator is used by the CI pipeline to automatically create a changelog entry for each new Loki release. Add other chart versions and respective changelog entries bellow this line."
```
Comment (MichelHollands) [inline production/helm/loki/CHANGELOG.md:14]: Can you leave the ( and ) ? This line is used by a tool to create changelog entries so it shouldn't be changed.

File: production/helm/loki/CHANGELOG.md [hunk 2]
```diff
@@ -11,7 +11,11 @@ Entries should be ordered as follows:
...
+
+## 5.39.0
```
Comment (MichelHollands) [inline production/helm/loki/CHANGELOG.md:16]: Thanks for adding this. You are still missing a change in the README.MD file though. See https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213 for an example. Without that change the CI will fail.

**Comment by MichelHollands** [comment]
@heytrav Can you run `make -C docs sources/setup/install/helm/reference.md` and add the changes? Once that is done the CI should pass and I'll merge it.

**Comment by heytrav** [comment]
> make -C docs sources/setup/install/helm/reference.md

I ran it like you said

```
❯ make -C docs sources/setup/install/helm/reference.md
make: Entering directory '/home/travis/repos/loki/docs'
make: 'sources/setup/install/helm/reference.md' is up to date.
make: Leaving directory '/home/travis/repos/loki/docs'
❯ git status
On branch swift-storage-config
Your branch is up to date with 'my-upstream/swift-storage-config'.

nothing to commit, working tree clean
```
It didn't actually make any changes

**Review by MichelHollands — APPROVED** [review]
LGTM

**Pull request merged**